sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuAggregateExec.scala

```diff
@@ -277,25 +279,32 @@ object AggregateUtils extends Logging {
 
       val newBuckets = batchesByBucket.flatMap(bucket => {
         if (needRepartitionAgain(bucket)) {
-          if (hashSeed + 7 > 200) {
-            log.warn("Too many times of repartition, may hit a bug? Size for each batch in " +
+          if (recursiveDepth >= maxRecursiveDepth) {
+            // Normally this should not happen, because we are repartitioning data that has
+            // already went through first round of aggregation, so there shouldn't be too many
+            // duplicated rows (the duplication only happens in different batches) to prevent
+            // repartitioning out (considering we're changing seed each time we repartition).
+            // However for some test cases with really small batch size, this can happen. So
+            // we're just logging some warnings here.
+            log.warn("The bucket is still too large after " + recursiveDepth +
```

Comment:
can we use  scala string interpolation here?

Reply:
It is also not clear to me why we end up in this situation in the tests. Can you give me an example?

Reply:
hi @abellina , if you replace this warning with a throw, and run ` DATAGEN_SEED=1732770215 SPARK_HOME=...... integration_tests/run_pyspark_from_build.sh -k 'hash_aggregate_test  and spark.rapids.sql.batchSizeBytes'` (the seed is same as in https://github.com/NVIDIA/spark-rapids/issues/11790#issuecomment-2505583503), then you'll get see failed tests like: 

![img_v3_02hl_004f86d4-c406-485b-956a-1279e4ad50eg](https://github.com/user-attachments/assets/be114459-5a90-4d76-a51c-662ab7cf8389)

Even if we change the max repartition number from 10 to 20, there're still failed test cases because of this.